src/comps/temp-withdraw.tsx

```diff
@@ -0,0 +1,240 @@
+      if (witnessProgram.length === 20) {
+        // P2WPKH
+        return { version: "0x04", hashbytes: hexHash };
+      } else if (witnessProgram.length === 32) {
+        // P2WSH
+        return { version: "0x05", hashbytes: hexHash };
+      } else {
+        throw new Error(
+          `Unsupported witness v0 length=${witnessProgram.length}`,
+        );
+      }
+    } else if (witnessVersion === 1 && witnessProgram.length === 32) {
+      // P2TR
+      return { version: "0x06", hashbytes: hexHash };
+    } else {
+      throw new Error(
+        `Unrecognized witnessVersion=${witnessVersion}, length=${witnessProgram.length}`,
+      );
+    }
+  }
+};
+
+const data: NameKeysInfo[] = [
+  {
+    nameKey: "address",
+    type: "text",
+    initValue: "",
+    placeholder: "address to receive your Bitcoin",
+  },
+  {
+    nameKey: "amount",
+    type: "number",
+    initValue: "",
+    placeholder: "withdraw amount",
+  },
+  {
+    nameKey: "fee",
+    type: "number",
+    initValue: "",
+    placeholder: "the max fee",
+  },
+];
+
+const BasicWithdraw = () => {
+  const [txId, setTxId] = useState<string | null>(null);
+  const { addresses, selectedWallet } = useAtomValue(walletInfoAtom);
+
+  const router = useRouter();
+
+  const { WALLET_NETWORK, SBTC_CONTRACT_DEPLOYER } =
+    useAtomValue(bridgeConfigAtom);
+
+  const handleSubmit = async (values: Record<string, string>) => {
+    console.log(values);
+    const { address, amount, fee } = values;
+
+    if (!WALLET_NETWORK) {
+      throw new Error("Invalid network");
+    }
+    if (!SBTC_CONTRACT_DEPLOYER) {
+      throw new Error("Invalid contract deployer");
+    }
+    // get the proper data for the address
+    const recipient = decodeBitcoinAddressToClarityRecipient(
+      address,
+      WALLET_NETWORK,
+    );
+
+    if (!recipient) {
+      throw new Error("Invalid recipient address");
+    }
+    const publicKey = addresses.stacks?.publicKey;
+
+    if (!publicKey) {
+      throw new Error("Invalid public key");
+    }
+    console.log(recipient);
+
+    // convert the amount to satoshis
+    const satoshiAmount = Math.round(parseFloat(amount) * 1e8);
+    console.log("satoshiAmount", satoshiAmount);
+
+    // convert the fee to satoshis
+    const satoshiFee = Math.round(parseFloat(fee) * 1e8);
+
+    const contractArgs = [
+      uintCV(satoshiAmount),
+      tupleCV({
+        version: bufferCV(hexToBytes(recipient.version)),
+        hashbytes: bufferCV(hexToBytes(recipient.hashbytes)),
+      }),
+      uintCV(satoshiFee),
+    ];
+
+    const stacksNetwork = getStacksNetwork(WALLET_NETWORK);
+
+    const transaction = await makeUnsignedContractCall({
+      contractAddress: SBTC_CONTRACT_DEPLOYER,
+      contractName: "sbtc-withdrawal",
+      functionName: "initiate-withdrawal-request",
+      functionArgs: contractArgs,
+      publicKey,
+      validateWithAbi: true,
+      network: stacksNetwork,
+      fee: WALLET_NETWORK === "mainnet" ? undefined : 10_000,
```

Comment:
qq: why do we need to specify the fee for non mainnet?

Reply:
the other environments sometimes fail in fee estimations and usually don't need more than 0.001 STX